azurerm/data_source_kubernetes_cluster.go

```diff
@@ -621,6 +639,18 @@ func flattenKubernetesClusterDataSourceLinuxProfile(input *containerservice.Linu
 	return []interface{}{values}
 }
 
+func flattenKubernetesClusterDataSourceWindowsProfile(input *containerservice.ManagedClusterWindowsProfile) []interface{} {
```

Comment:
so if this is a Linux cluster then this should be nil, but (below, line 651) we'll add an empty item to the state anyway; instead we can make this:

```
if input == nil {
  return []interface{}{}
}
```

which will ensure that the `windows_profile` block contains 0 elements when it's a Linux cluster